pkg/node/client.go

```diff
@@ -0,0 +1,165 @@
+	klog.Infof("getting bucketAccess %q", fmt.Sprintf("%s", baName))
+	ba, err := n.cosiClient.BucketAccesses().Get(ctx, baName, metav1.GetOptions{})
+	if err != nil {
+		return nil, logErr(getError("bucketAccess", fmt.Sprintf("%s", baName), err))
+	}
+	// TODO: need to enable validation after resolving status issue - Krish
+	//if ba == nil || !ba.Status.AccessGranted {
+	//	return nil, logErr(fmt.Errorf("BucketAccess is nil or BucketAccess does not grant access %q", fmt.Sprintf("%s", baName)))
+	//}
+	return ba, nil
+}
+
+func (n *NodeClient) getBR(ctx context.Context, brName, brNs string) (*v1alpha1.BucketRequest, error) {
+	klog.Infof("getting bucketRequest %q", brName)
+	br, err := n.cosiClient.BucketRequests(brNs).Get(ctx, brName, metav1.GetOptions{})
+	if err != nil {
+		return nil, logErr(getError("bucketRequest", fmt.Sprintf("%s/%s", brNs, brName), err))
+	}
+	// TODO: need to enable validation after resolving status issue - Krish
+	//if br == nil || !br.Status.BucketAvailable {
+	//	return nil, logErr(fmt.Errorf("BucketRequest is nil or BucketRequest is not available yet %q", fmt.Sprintf("%s/%s", brNs, brName)))
+	//}
+	return br, nil
+}
+
+func (n *NodeClient) getB(ctx context.Context, bName string) (*v1alpha1.Bucket, error) {
+	klog.Infof("getting bucket %q", bName)
+	// is BucketInstanceName the correct field, or should it be BucketClass
+	bkt, err := n.cosiClient.Buckets().Get(ctx, bName, metav1.GetOptions{})
+	if err != nil {
+		return nil, logErr(getError("bucket", bName, err))
+	}
+	// TODO: need to enable validation after resolving status issue - Krish
+	//if bkt == nil || !bkt.Status.BucketAvailable {
+	//	return nil, logErr(fmt.Errorf("Bucket is nil or Bucket is not available yet %q", fmt.Sprintf("%s", bName)))
+	//}
+	return bkt, nil
+}
+
+func (n *NodeClient) GetResources(ctx context.Context, barName, barNs string) (bkt *v1alpha1.Bucket, secret *v1.Secret, err error) {
+	bar, err := n.getBAR(ctx, barName, barNs)
+	if err != nil {
+		return
+	}
+
+	ba, err := n.getBA(ctx, bar.Spec.BucketAccessName)
+	if err != nil {
+		return
+	}
+
+	br, err := n.getBR(ctx, bar.Spec.BucketRequestName, barNs)
+	if err != nil {
+		return
+	}
+
+	bkt, err = n.getB(ctx, br.Spec.BucketInstanceName)
+	if err != nil {
+		return
+	}
+
+	secret, err = n.kubeClient.CoreV1().Secrets(barNs).Get(ctx, ba.Spec.MintedSecretName, metav1.GetOptions{})
+	if err != nil {
+		_ = logErr(getError("pod", fmt.Sprintf("%s/%s", barNs, ba.Spec.MintedSecretName), err))
+		return
+	}
+	return
+}
+
+func (n *NodeClient) getProtocol(bkt *v1alpha1.Bucket) (interface{}, error) {
+	var protocolConnection interface{}
+	var err error
+	klog.Infof("bucket protocol %+v", bkt.Spec.Protocol)
+	switch bkt.Spec.Protocol.Name {
+	case v1alpha1.ProtocolNameS3:
+		protocolConnection = bkt.Spec.Protocol.S3
+	case v1alpha1.ProtocolNameAzure:
+		protocolConnection = bkt.Spec.Protocol.AzureBlob
+	case v1alpha1.ProtocolNameGCS:
+		protocolConnection = bkt.Spec.Protocol.GCS
+	case "":
```

Comment:
can case "" be omitted and just combine with `default`?

Reply:
I don't think the "" and default cases are as important anymore since we have validation on the generated CRDs from the API so squashing them would make the code base cleaner.